helm-charts/splunk-otel-collector/templates/config/_common.tpl

```diff
@@ -216,6 +206,26 @@ resource/logs:
     {{- end }}
 {{- end }}
 
+{{/*
+The transform processor adds service.name attribute to logs the same way as it's done by istio for the generated traces
+https://github.com/istio/istio/blob/6237cb4e63cf9a332327cc0a815d6b46257e6f8a/pkg/config/analysis/analyzers/testdata/common/sidecar-injector-configmap.yaml#L110-L115
+This enables the correlation between logs and traces in Splunk Observability Cloud.
+*/}}
+{{- define "splunk-otel-collector.transformLogsProcessor" -}}
+transform/istio_service_name:
+  error_mode: ignore
+  log_statements:
+    - context: resource
+      statements:
+        - set(cache["owner_name"], attributes["k8s.pod.name"]) where attributes["k8s.pod.name"] != nil
+        # Name of the object owning the pod is taken from "k8s.pod.name" attribute by striping the pod suffix according
+        # to the k8s name generation rules (we don't want to put pressure on the k8s API server to get the owner name):
+        # https://github.com/kubernetes/apimachinery/blob/ff522ab81c745a9ac5f7eeb7852fac134194a3b6/pkg/util/rand/rand.go#L92-L127
+        - replace_pattern(cache["owner_name"], "^(.+?)-(?:(?:[0-9bcdf]+-)?[bcdfghjklmnpqrstvwxz2456789]{5}|[0-9]+)$$", "$$1") where cache["owner_name"] != nil
+        - set(attributes["service.name"], Concat([cache["owner_name"], attributes["k8s.namespace.name"]], ".")) where cache["owner_name"] != nil and attributes["k8s.namespace.name"] != nil
+        - set(attributes["service.name"], Concat([attributes["k8s.pod.labels.app"], attributes["k8s.namespace.name"]], ".")) where attributes["k8s.pod.labels.app"] != nil and attributes["k8s.namespace.name"] != nil
```

Comment:
`insert` for `service.name` was the existing process but not we're always overwriting?

Apart from that, since the update no longer relies on the dedicated `istio_service_name` attr and also restricts the app label setting to non-nil namespaces w/o a default I don't think these are equivalent, but it'd be hard to do that w/o a dedicated pipeline. I wonder if preserving `service.name` is enough though.

Reply:
> insert for service.name was the existing process but now* we're always overwriting?

Good catch! I'll revert it.

> Apart from that, since the update no longer relies on the dedicated istio_service_name attr and also restricts the app label setting to non-nil namespaces w/o a default I don't think these are equivalent, but it'd be hard to do that w/o a dedicated pipeline.

Yeah, I changed the logic a bit. I think we can assume the namespace is always available unless users explicitly remove it, in that case, we don't set anything because, if we set a default value, it'll be different from what's set by istio on traces.

Reply:
Updated to not override existing `service.name`